### PR TITLE
feat(core): Enable metrics for core package

### PR DESCRIPTION
**Pull request by renaynay**
While we do plan to remove the `bridge` node type eventually, it's useful to have some visibility into the core package in the short term.

### CODE REVIEW

File: core/metrics.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		m.blockTime = now
+	})
+}
+
+func (m *metrics) subscriptionStuck(ctx context.Context) {
+	m.observe(ctx, func(ctx context.Context) {
+		m.subscriptionStuckInst.Add(
+			ctx,
+			1,
+			metric.WithAttributes(attribute.String(subscriptionStuckTimestampKey, time.Now().String())),
```
Comment (Wondertan) [inline core/metrics.go:75]: Cardinality

File: core/metrics.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	subscriptionStuckInst metric.Int64Counter
+}
+
+func newMetrics() (*metrics, error) {
+	m := new(metrics)
+
+	var err error
+	m.blockTimeInst, err = meter.Float64Histogram(
+		"core_listener_block_time",
```
Comment (Wondertan) [inline core/metrics.go:28]: add _hist

File: core/metrics.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	if m == nil {
+		return
+	}
+
+	if ctx.Err() != nil {
+		ctx = context.Background()
+	}
+
+	observeFn(ctx)
+}
```
[the remark this answers is not in the document]
Comment (renaynay) [reply core/metrics.go:65]: can make a good first issue

**Review by Wondertan — CHANGES_REQUESTED** [review]
I realized that we don't need blockTime metric. The bridge node publishes headers through Subscriber, which goes through that validator, and time is tracked on it. So effectively we will get the same data twice

**Comment by renaynay** [comment]
@Wondertan the reason i wanted to capture block time here is that it's the only way to know on the DA-side what the approximate block time is so that we can see if there's a difference between core network block time and DA header propagation time. 

I removed it in this commit https://github.com/celestiaorg/celestia-node/pull/2863/commits/403bc9a924410d5675c652e16017cca5b049efc3

**Comment by Wondertan** [comment]
I understand and it turns out we designed things in a way that we don't need to duplicate anything 😺

**Comment by Wondertan** [comment]
We should aim to include fetcher metrics here as well

**Comment by Wondertan** [comment]
Linter btw

**Review by Wondertan — APPROVED** [review]
LGTM

**Review by distractedm1nd — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by ramin — APPROVED** [review]

**Pull request merged**